subprojects/core/src/main/java/org/gradle/internal/service/scopes/BuildSessionScopeServices.java

```diff
@@ -140,6 +140,13 @@ CrossProjectConfigurator createCrossProjectConfigurator(BuildOperationExecutor b
     ProjectCacheDir createCacheLayout(StartParameter startParameter, BuildLayoutFactory buildLayoutFactory, ProgressLoggerFactory progressLoggerFactory) {
         BuildLayout buildLayout = buildLayoutFactory.getLayoutFor(new BuildLayoutConfiguration(startParameter));
         File cacheDir = startParameter.getProjectCacheDir() != null ? startParameter.getProjectCacheDir() : new File(buildLayout.getRootDirectory(), ".gradle");
+        /*
+         * add cacheDir to default exclusions because not doing so causes and MD5 error
+         * when .gradle project cache is part of the root source code directory
+         * (typically in native builds) issue:
+         * https://github.com/gradle/gradle/issues/5941
+         */
+        DirectoryScanner.addDefaultExclude(cacheDir.getPath());
```

Comment:
:x: This needs to be cleaned up at the end of the build.

Reply:
Removed comments.